Galaxus.Functional/(Option)/(Extensions)/OptionLinqExtensions.cs

```diff
@@ -0,0 +1,143 @@
+        ///     range.
+        /// </summary>
+        /// <typeparam name="TSource">The type of the elements of <c>source</c>.</typeparam>
+        /// <param name="source">An <see cref="IEnumerable{T}" /> to return an element from.</param>
+        /// <param name="index">The zero-based index of the element to retrieve.</param>
+        /// <returns>
+        ///     <c><see cref="Option{T}" />.None</c> if the index is outside the bounds of the source sequence; otherwise, the
+        ///     element at the specified position in the source sequence.
+        /// </returns>
+        /// <exception cref="ArgumentNullException"><c>source</c> is <c>null</c>.</exception>
+        public static Option<TSource> ElementAtOrNone<TSource>(this IEnumerable<TSource> source, int index)
+        {
+            var value = source.ToList();
+            return index >= value.Count || index < 0 ? Option<TSource>.None : Option<TSource>.Some(value[index]);
+        }
+
+        /// <summary>
+        ///     Returns the first element of the sequence that satisfies a condition or <see cref="Option{T}.None" /> if the
+        ///     sequence contains no elements.
+        /// </summary>
+        /// <typeparam name="TSource">The type of the elements of <c>source</c>.</typeparam>
+        /// <param name="source">An <see cref="IEnumerable{T}" /> to return an element from.</param>
+        /// <returns>
+        ///     <c><see cref="Option{T}" />.None</c> if <c>source</c> is empty; otherwise, the first element in <c>source</c>.
+        /// </returns>
+        public static Option<TSource> FirstOrNone<TSource>(this IEnumerable<TSource> source)
+        {
+            var value = source.ToList();
+            return value.Count == 0 ? Option<TSource>.None : Option<TSource>.Some(value[0]);
+        }
+
+        /// <summary>
+        ///     Returns the first element of the sequence that satisfies a condition or <see cref="Option{T}.None" /> if the
+        ///     sequence contains no elements.
+        /// </summary>
+        /// <typeparam name="TSource">The type of the elements of <c>source</c>.</typeparam>
+        /// <param name="source">An <see cref="IEnumerable{T}" /> to return an element from.</param>
+        /// <param name="predicate">A function to test each element for a condition.</param>
+        /// <returns>
+        ///     <c><see cref="Option{T}" />.None</c> if <c>source</c> is empty or if no element passes the test specified by
+        ///     <c>predicate</c>; otherwise, the first element in <c>source</c> that passes the test specified by <c>predicate</c>.
+        /// </returns>
+        /// <exception cref="ArgumentNullException"><c>source</c> or <c>predicate</c> is <c>null</c>.</exception>
+        public static Option<TSource> FirstOrNone<TSource>(this IEnumerable<TSource> source,
+            Func<TSource, bool> predicate)
```

Comment:
I think this won't be too wide if done on a single line, but if not it every parameter should be on its own line at least ;)

Guess we should define an editorconfig or StyleCop analyzers...

Reply:
You're right, this is from my ReSharper config as I personally have quite a high zoom